### PR TITLE
Feature: New test server/api mock tool using markdown definitions

### PR DESCRIPTION
**Pull request by timotheeguerin**
Issue: Azure/autorest.testserver#246
Initial version of the new mock server. This provide a cli that can be used like this:
```
api-mock <include..>

Start the server

Positionals:
  include  List of wildcard pattern/folder to search for definition files.
                                                          [string] [default: []]

Options:
      --version  Show version number                                   [boolean]
      --help     Show help                                             [boolean]
  -v, --verbose  Run with verbose logging level.                       [boolean]
      --debug    Run with debug logging level.                         [boolean]
      --level    Run with given logging level.                          [string]
      --port     Port where to host the server          [number] [default: 3008]
```

For example to run in this repo: `api-mock ./routes`

Todo: 
- [x] Wire build/lint/test to ci

### CODE REVIEW

**Review by blaquezzz — APPROVED** [review]

File: packages/extensions/core/lib/pipeline/plugin-endpoint.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class AutoRestExtension extends EventEmitter {
       }
     });
     const writerProxy = new Writable({
-      write: (chunk: string | Buffer, encoding: string, callback: Function) => {
+      write: (chunk: string | Buffer, encoding: BufferEncoding, callback: Function) => {
```
Comment (daviwil) [inline packages/extensions/core/lib/pipeline/plugin-endpoint.ts:110]: Was this needed for the tool somehow?

**Review by daviwil — APPROVED** [review]
This is awesome, really nice work!

File: packages/apps/md-mock-api/examples/additionalProperties.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Additional properties
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply packages/apps/md-mock-api/examples/additionalProperties.md:1]: Sure, works for me. WIll update

File: packages/extensions/core/lib/pipeline/plugin-endpoint.ts (hunk 1 above)
Comment (timotheeguerin) [reply packages/extensions/core/lib/pipeline/plugin-endpoint.ts:110]: I had to upgrade typescript and rush require consistency in the typescript version. This is just a type change and `BufferEncoding` is an union of string type(`"utf-8" | "utf16" ...`)

Comment (daviwil) [reply packages/extensions/core/lib/pipeline/plugin-endpoint.ts:110]: Makes sense!

**Pull request merged**